src/flows/shared/steps/OrganizeCollection/Sidebar/SidebarNftIcon.tsx

```diff
@@ -114,6 +123,47 @@ function SidebarNftIcon({
     [handleNftLoaded, handleTokenRenderSuccess, token.dbid]
   );
 
+  const relayEnvironment = useRelayEnvironment();
+  useEffect(
+    function pollTokenWhileStillSyncing() {
+      const POLLING_INTERVAL_MS = 5000;
+      if (token.media?.__typename !== 'SyncingMedia') {
+        return;
+      }
+
+      let timeoutId: ReturnType<typeof setTimeout>;
+
+      async function refreshToken() {
+        await fetchQuery<SidebarNftIconPollerQuery>(
+          relayEnvironment,
+          graphql`
+            query SidebarNftIconPollerQuery($id: DBID!) {
+              tokenById(id: $id) {
+                ...SidebarNftIconFragment
+              }
+            }
+          `,
+          { id: token.dbid }
+        ).toPromise();
+
+        timeoutId = setTimeout(refreshToken, POLLING_INTERVAL_MS);
+      }
+
+      timeoutId = setTimeout(refreshToken, POLLING_INTERVAL_MS);
+
+      return () => clearTimeout(timeoutId);
+    },
+    [relayEnvironment, token.dbid, token.media?.__typename]
+  );
+
+  if (token.media?.__typename === 'SyncingMedia') {
+    return (
+      <LoadingContainer>
+        <LoadingText>Loading...</LoadingText>
+      </LoadingContainer>
+    );
+  }
+
```

Comment:
all of this is to handle when a user adds their tezos wallet and their tokens are still syncing. we want to show them a loading ui and refresh until their tokens show up

Reply:
1. is there any other case where this could possibly get triggered? maybe for new ETH users onboarding? but ETH uses opensea as a backup so the tokens may technically always be loaded
2. non-blocking, but would be great to see this in action with a fresh tezos wallet

Reply:
should be able to test tomorrow when benny gets back